miden-lib/asm/miden/note.masm

```diff
@@ -143,4 +143,171 @@ export.get_serial_number
 
     syscall.get_note_serial_number
     # => [SERIAL_NUMBER]
-end
\ No newline at end of file
+end
+
+#! Computes hash of note inputs starting at the specified memory address.
+#!
+#! This procedure divides the hashing process into two parts: hashing pairs of words using 
+#! `absorb_double_words_from_memory` procedure and hashing the remaining values using the `hperm`
+#! instruction. 
+#!
+#! Inputs:  [ptr, num_elements]
```

Comment:
We should use `inputs_ptr` and `num_inputs` because this procedure is specific to hashing inputs.